sdk/java-sdk-spring/src/main/scala/kalix/javasdk/client/ComponentCall.scala

```diff
@@ -0,0 +1,270 @@
+    val body = bodyIndex.map(params(_))
+
+    val kalixClientImpl = kalixClient.asInstanceOf[RestKalixClientImpl]
+
+    val pathTemplate = restMethod.parsedPath.path
+
+    requestMethod match {
+      case RequestMethod.GET =>
+        kalixClientImpl.runWithoutBody(HttpMethods.GET, pathTemplate, pathVariables, queryParams, returnType)
+      case RequestMethod.HEAD => notSupported(requestMethod, pathTemplate)
+      case RequestMethod.POST =>
+        kalixClientImpl.runWithBody(HttpMethods.POST, pathTemplate, pathVariables, queryParams, body, returnType)
+      case RequestMethod.PUT =>
+        kalixClientImpl.runWithBody(HttpMethods.PUT, pathTemplate, pathVariables, queryParams, body, returnType)
+      case RequestMethod.PATCH =>
+        kalixClientImpl.runWithBody(HttpMethods.PATCH, pathTemplate, pathVariables, queryParams, body, returnType)
+      case RequestMethod.DELETE =>
+        kalixClientImpl.runWithoutBody(HttpMethods.DELETE, pathTemplate, pathVariables, queryParams, returnType)
+      case RequestMethod.OPTIONS => notSupported(requestMethod, pathTemplate)
+      case RequestMethod.TRACE   => notSupported(requestMethod, pathTemplate)
+    }
+  }
+
+  private def getQueryParam(params: Seq[scala.Any], parameterIndex: Int): util.List[scala.Any] = {
+    val value = params(parameterIndex)
+    if (value == null) {
+      util.List.of()
+    } else if (value.isInstanceOf[util.List[_]]) {
+      value.asInstanceOf[util.List[scala.Any]]
+    } else {
+      util.List.of(value)
+    }
+  }
+
+  private def getPathParam(params: Seq[scala.Any], parameterIndex: Int, paramName: String): scala.Any = {
+    val value = params(parameterIndex)
+    if (value == null) {
+      throw new IllegalStateException(s"Path param [$paramName] cannot be null.")
+    }
+    value
+  }
+
+  private def notSupported(requestMethod: RequestMethod, pathTemplate: String) = {
+    throw new IllegalStateException(s"HTTP $requestMethod not supported when calling $pathTemplate")
+  }
+
+  private def idVariables(id: Option[String], method: Method): Map[String, String] = {
+
+    val declaringClass = method.getDeclaringClass
+    if (declaringClass.getAnnotation(classOf[EntityType]) == null &&
+      declaringClass.getAnnotation(classOf[TypeId]) == null) {
+      //not an entity or workflows
+      Map.empty
+    } else if (IdExtractor.shouldGenerateId(method)) {
+      Map.empty
+    } else {
+      val idNames = IdExtractor.extractIds(declaringClass, method)
+      id match {
+        case Some(value) => Map(idNames.head -> value) //TODO handle compound keys
+        case None        => throw new IllegalStateException(s"Id is missing while calling ${method.getName}")
+      }
+    }
+  }
+}
+
+// format: off
+final class ComponentCall2[A1, A2, R](kalixClient: KalixClient, lambda: scala.Any, entityId: Optional[String]) {
+
+  def params(a1: A1, a2: A2): DeferredCall[Any, R] = {
```

Comment:
I agree that it's not necessary to add a detailed javadoc for all the 20 params method variants, but we can find some generic, like:

```java
/**
 * Pass in the parameters that are required to execute this call. 
 * 
 * The types and order of parameters are the same as defined in the method reference 
 * used to build this ComponentCall.
 */

```

And then we can repeat it each time with not further detail.

Reply:
added